core/src/main/java/org/infinispan/reactive/publisher/impl/ClusterPublisherManagerImpl.java

```diff
@@ -861,6 +920,13 @@ public <R> SegmentPublisherSupplier<R> entryPublisher(IntSet segments, Set<K> ke
                                              deliveryGuarantee, batchSize, transformer);
    }
 
+   @Override
+   public CompletionStage<Long> sizePublisher(IntSet segments, InvocationContext ctx, long flags) {
+      long addedFlags = EnumUtil.mergeBitSets(flags, FlagBitSets.CACHE_MODE_LOCAL);
```

Comment:
This flag shouldn't be set here, this should be doing an entire cluster size check. TBH I would expect this to cause a lot of test failures.